master/internal/db/postgres.go

```diff
@@ -676,6 +676,27 @@ WHERE id = $1`, &experiment, id); err != nil {
 	return &experiment, nil
 }
 
+// ExperimentWithoutBackwardsIncompatibleFieldsByID looks up an experiment by ID in a database,
+// returning an error if none exists.
+// TODO(DET-4009): Remove when we have a better story for backwards compatibility.
+func (db *PgDB) ExperimentWithoutBackwardsIncompatibleFieldsByID(
+	id int,
+) (*model.Experiment, error) {
+	var experiment model.Experiment
+
+	if err := db.query(`
+SELECT id, state,
+	   config #- '{searcher}' #- '{min_validation_period}' #- '{min_checkpoint_period}' AS config,
+	   model_definition, start_time, end_time, archived,
+       git_remote, git_commit, git_committer, git_commit_date, owner_id
```

Comment:
non-blocking: line indent seems off

Reply:
yeah, I think it's mixed tabs/spaces and how github renders them. i'll fix.